### PR TITLE
Add support for update entity's update_percentage state attribute

### PR DESCRIPTION
**Pull request by emontnemery**
## Proposed change

Add support for update entity's `update_percentage` state attribute

Backend PR: https://github.com/home-assistant/core/pull/128877

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by piitaya — DISMISSED** [review]
Do we need to have backward compatibility or not? Entities that are not migrated yet to the new format will not have percentage.
I'm ok with that but I wanted to be sure. Otherwise, the PR is ok for me.

**Comment by emontnemery** [comment]
@piitaya there's no need for backwards compatibility in frontend, the core `update` integration makes sure that:
- The`in_progress` state attribute is `True` if the update entity's `in_progress` property is `True`, or a number
- The `update_percentage` state attribute is a number if the update entity's `in_progress` property is a number, otherwise it's `None` /  `null`

File: src/data/update.ts [hunk 1]
```diff
@@ -24,12 +24,13 @@ export enum UpdateEntityFeature {
...
   auto_update: boolean | null;
   installed_version: string | null;
-  in_progress: boolean | number;
+  in_progress: boolean;
   latest_version: string | null;
   release_summary: string | null;
   release_url: string | null;
   skipped_version: string | null;
   title: string | null;
+  update_percentage: number | null;
```
Comment (bramkragten) [inline src/data/update.ts:33]: Will this attribute always be around, even when not updating of not supported?

Comment (emontnemery) [reply src/data/update.ts:33]: Yes, that's how it's implemented in the core PR

**Comment by bramkragten** [comment]
What is missing here is a change to the update screen in the voice assistant wizard, I'll add that tomorrow

**Review by piitaya — APPROVED** [review]

**Pull request merged**